### ISSUE TITLE
Migrate Multiclass Semantic Segmentation

### ISSUE THREAD

**Issue by VAM7686** [RoboJackets/urc-software#48]
## Description
Currently, the URC repository does not utilize machine learning of any kind for obstacle detection. A good way to kickstart our ML efforts would be to migrate the vision based perception strategy from the IGVC repository.

The vision based perception strategy in the IGVC repository relies on a [UNET](https://arxiv.org/abs/1505.04597) with an [EfficientNet](https://arxiv.org/abs/1905.11946) encoder to perform semantic segmentation on input images. The neural net for the IGVC competition was trained to classify pixels as line, barrel, or ground. For the URC, the network will probably have to be retrained to classify obstacles for the URC competition. Retraining will involve some effort to label images.

Our encoder is pretrained on the [ImageNet](https://en.m.wikipedia.org/wiki/ImageNet) dataset which allows us to utilize transfer learning to train our model. Networks trained from scratch need a massive amounts of data and compute. Even with the appropriate dataset and compute you would need to wait days or weeks to train such a model. Using this strategy we can get away with using 500ish images (versus millions of images needed if training from scratch) to fine tune the weights of the pretrained encoder and achieve highly accurate results.

## Requirements

 - Migrate [multiclass_segmentation.py](https://github.com/RoboJackets/igvc-software/blob/c4de30fc741c9fb3d0923596bb2263d947eec3dd/igvc_perception/src/multiclass_segmentation/multiclass_segmentation.py) and the corresponding [multiclass segmentation folder](https://github.com/RoboJackets/igvc-software/tree/master/igvc_perception/src/multiclass_segmentation) 
 - Migrate [multiclass_segmentation.launch](https://github.com/RoboJackets/igvc-software/blob/c4de30fc741c9fb3d0923596bb2263d947eec3dd/igvc_perception/launch/multiclass_segmentation.launch)
 - Successfully launches
